### ISSUE TITLE
PIL doesn't seem to handle the color map correctly in some color indexed TGA files

### ISSUE THREAD

**Issue by innovara** [python-pillow/Pillow#4830]
Please see my question in [Stack Overflow](https://stackoverflow.com/questions/63151906/is-there-any-other-attribute-that-affetcts-how-colors-are-displayed-on-an-indexe) where I explain everything in greater detail and more importantly, where Mark Setchell explains why he thinks there is a bug in PIL when handling some color indexed TGA files.

Basically, I am working with a sample color indexed TGA file to test a script that would invert its colors.

But before any of that, the issue is that PIL doesn't seem to handle the color map correctly when open it and the values of the color palette end up being wrong.

These are the sample files: [CCM8.TGA](https://www.fileformat.info/format/tga/sample/c36e10e31828483d832ef67561e20d5a/download) and [UCM8.TGA](https://www.fileformat.info/format/tga/sample/34402cd832c3456eb82628798f54709a/download).
And you will see that if you open show any of them:
```python
from PIL import Image
Image.open('CCM8.TGA').show()
```
PIL shows:
![pQ6IH](https://user-images.githubusercontent.com/1219020/89067243-ce54a480-d366-11ea-91f1-f7c95b843d27.png)
When it should be:
![IpPCy](https://user-images.githubusercontent.com/1219020/89067319-f512db00-d366-11ea-9af7-e7c813be8669.png)

As I said, I wrote all the details on that post. If you want me to submit more information here, please let me know.

I am using Pillow 7.0 on Fedora 32 and Python 3.8.

**Comment by radarhere** [comment]
Do you have an image that demonstrate this problem, that are able to be added to the test suite under the Pillow license?

**Comment by innovara** [comment]
I have recreated those images which you can add to the test suite.

[TGA_samples.zip](https://github.com/python-pillow/Pillow/files/5013124/TGA_samples.zip)

Thank you.

**Comment by radarhere** [comment]
I've created #5396 to fix an unrelated bug mentioned in the StackOverflow answer.
I've created #5400 to fix the reading in of colors.
You would think that would be the end of it, but saving straightaway to PNG (`show()` uses the PNG format) is actually another problem again, and I think that won't be fixed until we significantly improve ImagePalette.

If #5400 is merged, then the following will work.
```python
from PIL import Image
Image.open('UCM8.TGA').convert("RGB").show()
```
